### PR TITLE
Async telemetry

### PR DESCRIPTION
**Pull request by maciekpaprocki**
Major redesign of telemetry plugin
- Added new acf field for async select
- Removed calls to telemetry on load field (ajax call is made iff you are on post-edit and you have minimum one telemetry widget)
- Improved security, all the calls are now proxied through wordpress and api token is not visible
- Added download buttons inside of widget
- Removing option to add more than one telemetry widget
- Fixed issues with other post types
- Moved all variables to env variables (BREAKING CHANGE!, need to be set up first)

### CODE REVIEW

**Comment by maciekpaprocki** [comment]
Tests are added in https://github.com/shortlist-digital/croissant-heavy/tree/telemetry-plugin, once approved we can merge everything together

**Comment by jonsherrard** [comment]
> Moved all variables to env variables (BREAKING CHANGE!, need to be set up first)
@maciekpaprocki are these documented anywhere?

**Comment by maciekpaprocki** [comment]
Not documented but they are added in env checker (they will show warnings in admin panel). 

I will add script to env checker which will block deployment if variables are not properly set and add README here.

**Comment by jonsherrard** [comment]
@maciekpaprocki I've noticed the HandleDuplicate functionality has been removed - This is actually broken on the current Croissant Heavy setup as it is.

Is it possible to have, testing to ensure that when a user clicks "clone" in the post views, none of the Telemetry IDs are copied over.

Is there any provision in this code for handling when "clone post" is clicked?

**Comment by jonsherrard** [comment]
OK I've had a play with it and I think it's good to go once the Handle Duplicate functionality goes back in. @maciekpaprocki

**Comment by maciekpaprocki** [comment]
I added it back. 

Should I add tests for it? 
Although I know it's probably phrase that should be never said out loud, but I think it's so simple that it doesn't require tests :)

File: assets/ui/download.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        '<div class="acf-input" id="download_buttons_' + id + '">',
+        '</div>',
+        '</div>',
+        '</div>'
+    ].join('\n'));
+
+    const linksID = '#download_buttons_' + id;
+
+    function addButton(text, data) {
+        const $a = $('<a href="#">' + text + '</a><br>');
```
Comment (jonsherrard) [inline assets/ui/download.js]: can we add `class="acf-button button button-primary"`
and `style="margin-top: 10px;"` to these buttons,  they look a bit more `UI`-y

**Comment by jonsherrard** [comment]
It would be great to have a test - it's quite bad that this is broken on Croissant Heavy on Hyde at the moment - and it's broken a lot of promotions across ES and MH because of ID collisions. I'm going to add stronger validations to Telemetry as well.

**Comment by maciekpaprocki** [comment]
We have test suite including even client side testing (although quite basic as filling this big forms in selenium is nightmare). 
https://github.com/shortlist-digital/croissant-heavy/tree/telemetry-plugin/tests/Telemetry

**Pull request merged**